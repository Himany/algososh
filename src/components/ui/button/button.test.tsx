import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

describe('Button test', () => {
  it('Button with text', () => {
    const tree = renderer
      .create(<Button text='TEST TEXT'/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Button without text', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Button disabled', () => {
    const tree = renderer
      .create(<Button disabled={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Button is load', () => {
    const tree = renderer
      .create(<Button isLoader={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Button callback', () => {
    const textCallBacak = () => {alert('TEST')};
    window.alert = jest.fn();
    render(<Button text="test_button" onClick={textCallBacak}/>)
    const button = screen.getByText("test_button");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('TEST');
  }); 
});



