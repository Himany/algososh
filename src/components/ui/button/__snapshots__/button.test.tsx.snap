// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test Button is load 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    TEST TEXT
  </p>
</button>
`;

exports[`Button test Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
